rust_src/src/buffers.rs

```diff
@@ -1144,7 +1144,7 @@ pub fn generate_new_buffer_name(name: LispStringRef, ignore: LispObject) -> Lisp
         let mut s = format!("<{}>", suffix_count);
         local_unibyte_string!(suffix, s);
         let candidate = unsafe { concat2(basename, suffix) };
-        if string_equal(candidate, ignore)
+        if string_equal(candidate.into(), ignore.into())
             || get_buffer(LispBufferOrName::Name(candidate)).is_none()
```

Comment:
See of `LispBufferOrName` uses the `Name` enum value. We know `candidate` is a string at this point. No need to go through the tests for is Symbol.

Reply:
What tests for is symbol are you referring to?

Reply:
Object (which we know is a string) -> LispSymbolOrString -> LispStringRef 

In the From<LispObject> there is this: `if o.is_string() || o.is_symbol() {` and then in the From<LispSymbolOrString> for LispStringRef there is this: `match s.0.as_symbol()`. Even after passing `is_string` we again try to make the value a symbol. And enum type would avoid some of this duplication or potentially do it more cheaply.